framework/helpers/BaseHtml.php

```diff
@@ -1719,7 +1719,7 @@ public static function renderSelectOptions($selection, $items, &$tagOptions = []
         );
         $options['encode'] = ArrayHelper::getValue($options, 'encode', $encode);
 
-        if ($prompt = ArrayHelper::remove($tagOptions, 'prompt')) {
+        if (null !== ($prompt = ArrayHelper::remove($tagOptions, 'prompt'))) {
```

Comment:
and null !== is to be avoided. Should match the other way around or ideally use isset() for readibility (if possible)